src/tmpfiles/tmpfiles.c

```diff
@@ -715,7 +715,7 @@ static int dir_cleanup(
                                         continue;
                                 }
 
-                                if (flock(dirfd(sub_dir), LOCK_EX|LOCK_NB) < 0) {
+                                if (flock(dirfd(sub_dir), LOCK_SH|LOCK_NB) < 0) {
```

Comment:
Hmm, generally, one would take a shared lock for reading and non-destructive atomic changes, because it's fine if there are multiple such users. For writing and destructive changes, an exclusive lock.

52b32b2aac96562ae8c380d8869b667f806471e0 says "[clients] can simply take a BSD file lock on any directory they like and thus exclude it from automatic aging". But if we take a shared lock, that doesn't work.